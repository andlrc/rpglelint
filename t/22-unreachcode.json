[
   {
      "column":5,
      "file":"t/22-unreachcode.rpgle",
      "line":15,
      "message":"code will never be executed",
      "severity":"warning"
   }
]
