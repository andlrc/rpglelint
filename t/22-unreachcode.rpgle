// rpglelint: -Wunreachable-code

dcl-proc proc1;
  if a > 1;
  endif;

  b = 1;

  return;
end-proc;

dcl-proc proc2;
  dow a > 1;
    iter;
    a = 1;
  enddo;
end-proc;

dcl-proc proc3;
  exsr a;
  a = 1;

  begsr a;
    dow a > 1;
      return 1;
    enddo;
  endsr;
end-proc;

dcl-proc proc4;
  exsr a;
  a = 1;

  begsr a;
    dow a > 1;
      leave;
    enddo;
  endsr;
end-proc;
