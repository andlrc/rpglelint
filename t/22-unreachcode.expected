t/22-unreachcode.rpgle:15:5: warning: code will never be executed [-Wunreachable-code]
     a = 1;
     ^
